### PR TITLE
Populate ThemeVideoIds and ThemeSongIds

### PR DESCRIPTION
**Pull request by cvium**
**Changes**
Extract theme songs and videos from ExtraIds when requested.

### CODE REVIEW

File: MediaBrowser.Controller/Entities/BaseItem.cs [hunk 1]
```diff
@@ -2909,7 +2950,7 @@ public bool UpdateRatingToItems(IList<BaseItem> children)
 
         public IEnumerable<BaseItem> GetThemeSongs()
         {
-            return ThemeVideoIds.Select(LibraryManager.GetItemById).Where(i => i.ExtraType.Equals(Model.Entities.ExtraType.ThemeSong)).OrderBy(i => i.SortName);
+            return ThemeSongIds.Select(LibraryManager.GetItemById).Where(i => i.ExtraType.Equals(Model.Entities.ExtraType.ThemeSong)).OrderBy(i => i.SortName);
```
Comment (Bond-009) [inline MediaBrowser.Controller/Entities/BaseItem.cs:2953]: `Where(i => i.ExtraType.Equals(Model.Entities.ExtraType.ThemeSong))` this shouldn't be needed, no?

Comment (cvium) [reply MediaBrowser.Controller/Entities/BaseItem.cs:2953]: No, I forgot to remove it

**Comment by crobibero** [comment]
Shouldn't the items still be ordered?

**Comment by cvium** [comment]
They are ordered by SortName in GetExtras()

**Comment by cvium** [comment]
Maybe the ordering is overwritten temporarily when scanning since it overwrites the entire array with new updates.

**Review by crobibero — APPROVED** [review]

**Review by Ullmie02 — APPROVED** [review]

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**